### PR TITLE
fixed javaValue generation for StringArray class

### PR DESCRIPTION
**Pull request by daudich**
Important fixes are on lines 1035-1044. The rest of the changes are style fixes from ESLint.

### CODE REVIEW

**Review by kirkeaton — APPROVED** [review]

File: src/foam/java/refinements.js [hunk 1]
```diff
@@ -10,7 +10,9 @@ foam.CLASS({
     {
       class: 'String',
       name: 'javaType',
-      factory: function() { return this.of; }
+      factory: function() {
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:13]: Why this change?

File: src/foam/java/refinements.js [hunk 2]
```diff
@@ -96,25 +100,25 @@ foam.CLASS({
...
-        propType:         this.javaType,
-        propValue:        this.javaValue,
-        propRequired:     this.required,
-        cloneProperty:    this.javaCloneProperty,
-        diffProperty:     this.javaDiffProperty,
-        jsonParser:       this.javaJSONParser,
-        queryParser:      this.javaQueryParser,
-        csvParser:        this.javaCSVParser,
-        extends:          this.javaInfoType,
+        sourceCls: cls,
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:103]: Why this change?

File: src/foam/java/refinements.js [hunk 3]
```diff
@@ -294,16 +298,18 @@ foam.LIB({
 
       var axioms = this.getOwnAxioms();
 
-      for ( var i = 0 ; i < axioms.length ; i++ ) {
+      for ( var i = 0; i < axioms.length; i ++ ) {
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:301]: space before ;'s, no space before ++.

File: src/foam/java/refinements.js [hunk 4]
```diff
@@ -518,7 +539,9 @@ foam.CLASS({
       getter: function() {
         // TODO: This could be an expression if the copyFrom in createChildMethod
         // didn't finalize its value
-        if ( this.name == 'find' ) console.log(this.name, "returns", this.javaReturns)
+        if ( this.name == 'find' ) {
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:542]: Why?

File: src/foam/java/refinements.js [hunk 5]
```diff
@@ -1399,8 +1439,8 @@ foam.CLASS({
       name: 'javaImports',
       adaptArrayElement: function(o) {
         return typeof o === 'string' ?
-          foam.java.JavaImport.create({import: o}) :
-          foam.java.JavaImport.create(o) ;
+          foam.java.JavaImport.create({ import: o }) :
```
Comment (kgrgreer) [inline src/foam/java/refinements.js:1442]: Why?

File: src/foam/java/refinements.js (hunk 4 above)
Comment (daudich) [reply src/foam/java/refinements.js:542]: The string was longer than 80 characters.

File: src/foam/java/refinements.js (hunk 1 above)
Comment (daudich) [reply src/foam/java/refinements.js:13]: Fixed: https://github.com/foam-framework/foam2/pull/1239

File: src/foam/java/refinements.js (hunk 2 above)
Comment (daudich) [reply src/foam/java/refinements.js:103]: Fixed: https://github.com/foam-framework/foam2/pull/1239

File: src/foam/java/refinements.js (hunk 3 above)
Comment (daudich) [reply src/foam/java/refinements.js:301]: Fixed for all `for`s: https://github.com/foam-framework/foam2/pull/1239

File: src/foam/java/refinements.js (hunk 5 above)
Comment (daudich) [reply src/foam/java/refinements.js:1442]: Fixed: https://github.com/foam-framework/foam2/pull/1239
